### ISSUE TITLE
terminal issues for both vscode and vscode insiders with zsh p10k zsh-autosuggestion

### ISSUE THREAD

**Issue by Norfeldt** [microsoft/vscode#166936]
Does this issue occur when all extensions are disabled?: Yes/No

# VS Code Version: 

- Version: 1.73.1 (Universal)
- Commit: 6261075646f055b99068d3688932416f2346dd3b
- Date: 2022-11-09T02:08:38.961Z
- Electron: 19.0.17
- Chromium: 102.0.5005.167
- Node.js: 16.14.2
- V8: 10.2.154.15-electron.0
- OS: Darwin arm64 22.1.0
- Sandboxed: No

# OS Version: 
- Ventura 13.0.1
- Apple M1 Chip

vscode (stable)

![Nov-22-2022 09-11-27](https://user-images.githubusercontent.com/678806/203260100-f8815096-0ad1-4f48-81cf-220cd9e464f0.gif)

ohmyzsh with p10k and zsh-autosuggestions behaves really weird

> .zshrc

```sh
# Enable Powerlevel10k instant prompt. Should stay close to the top of ~/.zshrc.
# Initialization code that may require console input (password prompts, [y/n]
# confirmations, etc.) must go above this block; everything else may go below.
if [[ -r "${XDG_CACHE_HOME:-$HOME/.cache}/p10k-instant-prompt-${(%):-%n}.zsh" ]]; then
  source "${XDG_CACHE_HOME:-$HOME/.cache}/p10k-instant-prompt-${(%):-%n}.zsh"
fi
source /usr/local/opt/powerlevel10k/powerlevel10k.zsh-theme
# To customize prompt, run `p10k configure` or edit ~/.p10k.zsh.
[[ ! -f ~/.p10k.zsh ]] || source ~/.p10k.zsh
typeset -g POWERLEVEL9K_INSTANT_PROMPT=quiet


# Path to your oh-my-zsh installation.
export ZSH="/Users/norfeldt/.oh-my-zsh"

DISABLE_UPDATE_PROMPT="true" # Uncomment the following line to automatically update without prompting.
export UPDATE_ZSH_DAYS=13

setopt autocd
zstyle ':completion:*' menu select
zstyle ':completion:*' matcher-list 'r:|[._-]=* r:|=*' 'l:|=* r:|=*' 'm:{a-zA-Z}={A-Za-z}'
autoload -U compinit
compinit

# History (Heather setup)
if [ -z "$HISTFILE" ]; then
  HISTFILE=$HOME/.zsh_history
fi

HISTSIZE=1000
SAVEHIST=1000

setopt EXTENDED_HISTORY
setopt HIST_EXPIRE_DUPS_FIRST
setopt HIST_IGNORE_DUPS
setopt HIST_IGNORE_ALL_DUPS
setopt HIST_IGNORE_SPACE
setopt HIST_FIND_NO_DUPS
setopt HIST_SAVE_NO_DUPS
setopt HIST_BEEP

source ~/.oh-my-zsh/plugins/git/git.plugin.zsh
source ~/.oh-my-zsh/plugins/autojump/autojump.plugin.zsh
source ~/.oh-my-zsh/plugins/command-not-found/command-not-found.plugin.zsh
source ~/.zsh/zsh-autosuggestions/zsh-autosuggestions.zsh
source ~/zsh-syntax-highlighting/zsh-syntax-highlighting.zsh
ZSH_HIGHLIGHT_STYLES[path]='fg=cyan,bold'
ZSH_HIGHLIGHT_STYLES[alias]='fg=green,underline'

[[ -s `brew --prefix`/etc/autojump.sh ]] && . `brew --prefix`/etc/autojump.sh

# Plugins
plugins=(
  git
  autojump
  command-not-found
  zsh-autosuggestions
  zsh-syntax-highlighting
  history
)
```

The **vscode insiders** seems to have fixed that, but is having some bad issues displaying emojis and other special characters

<img width="609" alt="Screenshot 2022-11-22 at 08 25 59" src="https://user-images.githubusercontent.com/678806/203261180-2bc63c29-d60f-4547-82e4-bb096b61be9f.png">

This is how it looks on vscode (stable):

<img width="753" alt="Screenshot 2022-11-22 at 08 28 05" src="https://user-images.githubusercontent.com/678806/203261392-ef84aa5a-39dc-425a-bd33-5aa2855b1dd4.png">

**Comment by ttdatt** [comment]
Yes, I upgrade vscode 1.74.0 this morning, and suddenly zsh-autosuggestion stop working

Version: 1.74.0
Commit: 5235c6bb189b60b01b1f49062f4ffa42384f8c91
Date: 2022-12-05T16:37:54.581Z
Electron: 19.1.8
Chromium: 102.0.5005.167
Node.js: 16.14.2
V8: 10.2.154.15-electron.0
OS: Darwin arm64 22.1.0
Sandboxed: No

**Comment by meganrogge** [comment]
I was able to reproduce this at one point, but cannot in Insider's now

![Screenshot 2022-12-13 at 4 11 29 PM](https://user-images.githubusercontent.com/29464607/207455021-80f9112e-be94-4c41-829b-b50ea526a70a.png)

**Comment by meganrogge** [comment]
works for Chinese and other chars now too

![Screenshot 2022-12-13 at 4 16 33 PM](https://user-images.githubusercontent.com/29464607/207456253-dc3194fd-5402-4300-b02f-360252b2a19c.png)

**Comment by pepelsbey** [comment]
Still broken for me in 1.75.0 Insider, empty profile: trying to type йцукен (Cyrillic)

https://user-images.githubusercontent.com/105274/207462292-223eb41c-be19-4f33-a72c-9871c92a0175.mov

**Comment by meganrogge** [comment]
what's the value of `$LANG` for you @pepelsbey

**Comment by meganrogge** [comment]
this reproduces when `$LANG` is explicitly set as determined by @rzhao271

**Comment by pepelsbey** [comment]
> what's the value of `$LANG` for you @pepelsbey

@meganrogge en_DE.UTF-8

**Comment by rzhao271** [comment]
Confirmation steps on macOS:
1. Run the "Configure Runtime Arguments" command. Make sure `argv.json` doesn't have a locale key-value pair
2. Set the system language to Simplified Chinese and restart the system
3. Quit and reopen VS Code
4. Try pasting the phrase 测试 into the terminal (zsh or bash)
5. :bug: The characters are not rendered properly

**Comment by Norfeldt** [comment]
> Yes, I upgrade vscode 1.74.0 this morning, and suddenly zsh-autosuggestion stop working
> 
> Version: 1.74.0 Commit: [5235c6b](https://github.com/microsoft/vscode/commit/5235c6bb189b60b01b1f49062f4ffa42384f8c91) Date: 2022-12-05T16:37:54.581Z Electron: 19.1.8 Chromium: 102.0.5005.167 Node.js: 16.14.2 V8: 10.2.154.15-electron.0 OS: Darwin arm64 22.1.0 Sandboxed: No
> 
> update: @Norfeldt disable `terminal.integrated.shellIntegration.enabled` and it'll be alright

@ttdatt I did:


```
  "terminal.integrated.shellIntegration.enabled": false,
 ```
 
 and my stable version (which has updated since reporting the issue) looks like this:

<img width="479" alt="Screenshot 2022-12-14 at 14 03 44" src="https://user-images.githubusercontent.com/678806/207602526-fcd5c5b6-4bc0-4e90-a69c-3a1f700602cf.png">

**Comment by meganrogge** [comment]
Before the change:
![截屏2022-12-14 上午11 49 58](https://user-images.githubusercontent.com/29464607/207671859-e1479c4c-06c4-4b4b-933e-37b29dd26582.png)

After the change:
![截屏2022-12-14 上午11 54 34](https://user-images.githubusercontent.com/29464607/207671863-18687f43-0dca-4db7-803d-b1581fd81ce1.png)

**Comment by rzhao271** [comment]
@Norfeldt what do you get when you type the command `echo $LANG` into the terminal? My fix does not fix the issue for all locales&mdash;it just fixes a regression that broke even more locales.

**Comment by Hazlank** [comment]
I install Language pack extension for Chinese, then the terminal can display chinese characters correct

**Comment by Norfeldt** [comment]
@rzhao271 it's `en_SE.UTF-8`
